views/default/create_packages.pdt

```diff
@@ -0,0 +1,465 @@
+                                <tr class="heading_row">
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.tld');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.currency');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.previous_registration_price');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.previous_renewal_price');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.current_registration_price');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.current_renewal_price');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.price');?></td>
+                                    <td class="last"><?php $this->_('Namesilo.manage.create_packages_heading.renew_price');?></td>
+                                </tr>
+                            </thead>
+                            <tbody>
+                            </tbody>
+                        </table>
+                    </li>
+                </ul>
+            </div>
+
+            <div class="title_row">
+                <h3><?php $this->_('Namesilo.manage.create_packages.heading_group');?></h3>
+            </div>
+            <div class="pad">
+                <ul>
+                    <li>
+                        <?php
+                        $this->Form->label($this->_('Namesilo.manage.create_packages.field_package_group', true), 'package_group');
+                        $this->Form->fieldSelect('package_group', $package_groups, $this->Html->ifSet($vars->package_group));
+                        ?>
+                    </li>
+                </ul>
+            </div>
+        </div>
+
+        <div class="inner">
+            <div class="button_row">
+                <?php
+                $this->Form->fieldSubmit('save', $this->_('Namesilo.manage.create_packages.create_packages_btn', true), ['class' => 'btn btn-primary pull-right']);
+                ?>
+            </div>
+        </div>
+        <?php
+        $this->Form->end();
+
+        $this->Widget->end();
+        ?>
+
+        <script type="text/javascript" src="<?php echo VENDORWEBDIR . 'blesta/ckeditor/ckeditor.js';?>"></script>
+        <script type="text/javascript" src="<?php echo VENDORWEBDIR . 'blesta/ckeditor/adapters/jquery.js';?>"></script>
+        <script type="text/javascript">
+            $(document).ready(function () {
+                // Load TLD rows
+                <?php
+                $url = explode('?', $_SERVER['REQUEST_URI'], 2);
+                ?>
+                $('input.tld').prop('disabled', true);
+
+                $('#tld_rows tbody').load('<?php echo $this->Html->ifSet($url[0]);?>?action=tld_rows', function() {
```

Comment:
So I'm curious as to what is doable here.  This still takes a while to load and loads a bunch of html which makes the jquery operations pretty dang slow.  Would it make sense, rather than loading all at once here, instead loading each set of tld rows when a checkbox is selected.  It would require a request to the server each time, but the pricing from the API is cached, yes?  So we would not be making any additional API calls?

If that is not possible or doesn't make sense, then we might want to add some kind of message about the wait time that disappears after the tld rows are loaded.  When you first load the page you just see a bunch of disabled checkboxes that you can't interact with for the first ten second and you have no clue why.

Reply:
Yes, the API from the API is cached. I modified the view to load only the TLDs that are enabled and the loading time improved dramatically. I need to make some additional tests to check that no changes are needed in namesilo_packages.php due to this modification.